examples/wallet/app/actions/account.js

```diff
@@ -42,11 +44,30 @@ export function setAccount(privateKey: string): Thunk<SetKeysAction> {
   };
 }
 
-const initializeKeysAndRedirect = (dispatch, keys: AccountKeys) => {
+export function loadAccountFromPrivateKey(
+  privateKey: string
+): Thunk<SetKeysAction> {
+  return function setAccountThunk(dispatch) {
+    return getAccountFromPrivateKey(privateKey).then(loadedPrivateKey =>
+      curry(initializeKeysAndRedirect)(dispatch, loadedPrivateKey, false)
+    );
+  };
+}
+
+const initializeKeysAndRedirect = (
+  dispatch,
+  keys: AccountKeys,
+  saveAccount = true
```

Comment:
do optional parameters count towards function arity? It'd be nice to have a comment explianing how currying it works, or avoid the optional parameter altogether